### PR TITLE
Replace `FileUtils.chmod` with `/bin/chmod` for cask binaries.

### PR DESCRIPTION
**Pull request by commitay**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [ ] Have you successfully run `brew tests` with your changes locally?

-----
Hello Homebrew maintainers, this is my first PR here, apologies for it being incomplete at this stage.

Currently when creating cask binary symlinks `FileUtils.chmod` fails with some casks.

This fixes https://github.com/caskroom/homebrew-cask/issues/31874.

This is a WIP as `binary_spec.rb` is not included in this PR so tests will fail.

CC @reitermarkus

### CODE REVIEW

File: Library/Homebrew/cask/lib/hbc/artifact/binary.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ def install_phase
 
       def link
         super
-        FileUtils.chmod "+x", source
+        @command.run!("/bin/chmod", args: ["+x", source])
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/artifact/binary.rb]: This should be handled in a similar way as https://github.com/Homebrew/brew/blob/a365e16b48c5ee3461fdda48d8aca58df954c4eb/Library/Homebrew/cask/lib/hbc/artifact/moved.rb#L35-L39.

Also, this should be skipped if it is already executable (`source.executable?`).

Comment (commitay) [inline Library/Homebrew/cask/lib/hbc/artifact/binary.rb]: Sorry, I missed the notification. I probably should have mentioned in the HBC issue that I don't actually know ruby (or programming in general). I'm not sure how to skip the command if it is already executable.
```
  if source.writable?
    FileUtils.chmod "+x", source
  else
    SystemCommand.run("/bin/chmod", args: ["+x", source], sudo: true))
  end
```

Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/artifact/binary.rb]: To skip it, simply return out of the method:

```ruby
return if source.executable?
```

And `SystemCommand.run` should still be `@command.run!`, other than that this looks good.

Comment (commitay) [inline Library/Homebrew/cask/lib/hbc/artifact/binary.rb]: Like this or have I got it wrong:
```
return if source.executable?
  if source.writable?
    FileUtils.chmod "+x", source
  else
    @command.run!.run("/bin/chmod", args: ["+x", source], sudo: true))
  end
end
```

Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/artifact/binary.rb]: > I don't actually know ruby (or programming in general).

Well, then I guess it's about time. 😄

Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/artifact/binary.rb]: > Like this or have I got it wrong:

Almost, you have to remove last `end`, the `return` belongs to the `if`, but the rest does not.

Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/artifact/binary.rb]: Also, there's an extra bracket at the end of the `@command` line.

Comment (commitay) [inline Library/Homebrew/cask/lib/hbc/artifact/binary.rb]: > Also, there's an extra bracket at the end of the @command line.

Thanks. Sorry for making you do a tutorial for a PR.

```require "hbc/artifact/symlinked"

module Hbc
  module Artifact
    class Binary < Symlinked
      def install_phase
        super if CLI.binaries?
      end

      def link
        super
        return if source.executable?
          if source.writable?
            FileUtils.chmod "+x", source
          else
           @command.run!.run("/bin/chmod", args: ["+x", source], sudo: true)
          end
      end
    end
  end
end

Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/artifact/binary.rb]: Looks good, just need to fix the indentation on the last few lines (or run `brew style --fix`).

Comment (commitay) [inline Library/Homebrew/cask/lib/hbc/artifact/binary.rb]: I've run `brew style --fix` and the indentation is fixed so I'll update the PR.

I'm not sure if `binary_spec.rb` needs to be updated.

Thank you for you help.

**Comment by reitermarkus** [comment]
Thanks, @commitay!

**Comment by commitay** [comment]
Thank you for help and for finishing this @reitermarkus!

**Pull request merged**